### PR TITLE
add NRT collector

### PR DESCRIPTION
**Pull request by Garrybest**
Signed-off-by: Garrybest <garrybest@foxmail.com>

#### Which issue(s) this PR fixes:

Fixes #

### CODE REVIEW

**Comment by chenkaiyue** [comment]
CI waterline has been approved

File: pkg/features/features.go [hunk 1]
```diff
@@ -33,6 +36,7 @@ var defaultFeatureGates = map[featuregate.Feature]featuregate.FeatureSpec{
 	CraneAutoscaling:           {Default: true, PreRelease: featuregate.Alpha},
 	CraneAnalysis:              {Default: true, PreRelease: featuregate.Alpha},
 	CraneNodeResource:          {Default: true, PreRelease: featuregate.Alpha},
+	CraneNodeResourceTopology:  {Default: true, PreRelease: featuregate.Alpha},
```
Comment (mfanjie) [inline pkg/features/features.go:39]: should we start from CraneNodeResourceTopology=false, as it requires to cooperate with crane-scheduler

File: pkg/ensurance/collector/collector.go [hunk 2]
```diff
@@ -216,6 +235,12 @@ func (s *StateCollector) UpdateCollectors() {
 			}
 		}
 	}
+	if utilfeature.DefaultFeatureGate.Enabled(features.CraneNodeResourceTopology) {
+		if _, exists := s.collectors.Load(types.NodeResourceTopologyCollectorType); !exists {
+			s.collectors.Store(types.NodeResourceTopologyCollectorType,
+				noderesourcetopology.NewNodeResourceTopology(s.nodeName, s.sysPath, s.nrtLister, s.nodeLister, s.kubeClient, s.craneClient))
```
Comment (mfanjie) [inline pkg/ensurance/collector/collector.go:241]: do you want to run this code repeatedly with a 10s interval? NewNodeResourceTopology is a one time thing, maybe can be moved to the caller and only trigger once?

File: pkg/features/features.go (hunk 1 above)
Comment (Garrybest) [reply pkg/features/features.go:39]: Sure.

File: pkg/ensurance/collector/collector.go (hunk 2 above)
Comment (Garrybest) [reply pkg/ensurance/collector/collector.go:241]: Got it.

**Review by mfanjie — APPROVED** [review]
LGTM

**Pull request merged**